### PR TITLE
Liquidity baking fixes

### PR DESCRIPTION
**Pull request by letier3110**

### CODE REVIEW

File: src/components/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -67,8 +67,20 @@ export const Dropdown = <T extends unknown>({
     <>
       <TouchableOpacity
         style={styles.valueContainer}
-        onPress={dropdownBottomSheetController.open}
-        onLongPress={onLongPress}
+        onPress={() => {
+          if (disabled) {
```
Comment (LendiHop) [inline src/components/dropdown/dropdown.tsx:71]: could you use conditional operator here?

File: src/components/header/generate-modal-options.util.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { StackNavigationOptions } from '@react-navigation/stack';
+import React, { FC } from 'react';
+
+import { HeaderCloseOptionsButton } from './header-close-options-button/header-close-options-button';
+
+export const generateModalOptions = (
```
Comment (unixvb) [inline src/components/header/generate-modal-options.util.tsx:6]: let's rename this function as it's used nor for modals.  mb `generateIntegratedAppOptions`

File: src/components/asset-amount-input/asset-amount-input.tsx [hunk 3]
```diff
@@ -99,7 +99,7 @@ export const AssetAmountInput: FC<AssetAmountInputProps> = ({
     <>
       <View style={styles.headerContainer}>
         <Label label={label} />
-        {hasExchangeRate && (
+        {hasExchangeRate && !hideUsdSwitcher && (
```
Comment (unixvb) [inline src/components/asset-amount-input/asset-amount-input.tsx:102]: remove `hideUsdSwitcher` property.  Token/Usd switcher should be available if `exchangeRate` is available.  Vova did approve this

File: src/components/wallet-address/wallet-address.tsx [hunk 4]
```diff
@@ -40,12 +41,12 @@ export const WalletAddress: FC<Props> = ({ publicKeyHash }) => {
         <TouchableOpacity
           style={styles.domainNameContainer}
           {...(isAndroid && { disallowInterruption: true })}
-          onPress={() => copyStringToClipboard(domainName)}
+          onPress={() => (noCopy === true ? 0 : copyStringToClipboard(domainName))}
```
Comment (LendiHop) [inline src/components/wallet-address/wallet-address.tsx:44]: onPress={noCopy === true ? undefined : () => copyStringToClipboard(domainName)}

mb its more convenient?

File: src/components/public-key-hash-text/public-key-hash-text.tsx [hunk 5]
```diff
@@ -8,14 +8,24 @@ import { usePublicKeyHashTextStyles } from './public-key-hash-text.styles';
 
 interface Props extends MarginProps {
   publicKeyHash: string;
+  noCopy?: boolean;
```
Comment (unixvb) [inline src/components/public-key-hash-text/public-key-hash-text.tsx:11]: the same with `disabled`

File: src/components/dropdown/dropdown.tsx [hunk 6]
```diff
@@ -20,6 +20,7 @@ export interface DropdownProps<T> {
 }
 
 export interface DropdownValueProps<T> {
+  isModal?: boolean;
```
Comment (unixvb) [inline src/components/dropdown/dropdown.tsx:23]: how does `isModal` prop related to shared `Dropdown` component?

File: src/components/button/button-large/button-large-primary/button-large-primary.styles.ts [hunk 7]
```diff
@@ -4,7 +4,7 @@ import { useButtonLargeSharedStyleConfig } from '../button-large.styles';
 export const useButtonLargePrimaryStyleConfig = createUseStylesConfig(({ colors }) => ({
   ...useButtonLargeSharedStyleConfig(),
   activeColorConfig: {
-    titleColor: colors.white,
+    titleColor: colors.black,
```
Comment (unixvb) [inline src/components/button/button-large/button-large-primary/button-large-primary.styles.ts:7]: this broke a lot of buttons 
image

File: src/components/button/button-large/button-large-white/button-large-white.styles.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { createUseStylesConfig } from '../../../../styles/create-use-styles';
+import { useButtonLargeSharedStyleConfig } from '../button-large.styles';
+
+export const useButtonLargeWhiteStyleConfig = createUseStylesConfig(({ colors }) => ({
+  ...useButtonLargeSharedStyleConfig(),
+  activeColorConfig: {
+    titleColor: '#ffffff',
```
Comment (unixvb) [inline src/components/button/button-large/button-large-white/button-large-white.styles.ts:7]: reuse `white` const

File: src/components/public-key-hash-text/public-key-hash-text.tsx [hunk 9]
```diff
@@ -8,20 +8,31 @@ import { usePublicKeyHashTextStyles } from './public-key-hash-text.styles';
...
+      e.stopPropagation();
+      copyStringToClipboard(publicKeyHash);
+    }
   };
 
   return (
     <TouchableOpacity
       style={[styles.container, { marginTop, marginRight, marginBottom, marginLeft }]}
       onPress={handlePress}
+      disabled={disabled}
```
Comment (unixvb) [inline src/components/public-key-hash-text/public-key-hash-text.tsx:35]: event handlers should be last props

File: src/components/public-key-hash-text/public-key-hash-text.tsx [hunk 10]
```diff
@@ -8,20 +8,31 @@ import { usePublicKeyHashTextStyles } from './public-key-hash-text.styles';
...
+}) => {
   const styles = usePublicKeyHashTextStyles();
 
   const handlePress = (e: GestureResponderEvent) => {
-    e.stopPropagation();
-    copyStringToClipboard(publicKeyHash);
+    if (disabled !== true) {
+      e.stopPropagation();
+      copyStringToClipboard(publicKeyHash);
+    }
```
Comment (unixvb) [inline src/components/public-key-hash-text/public-key-hash-text.tsx:28]: do you need this condition with `disabled` `TouchableOpacity`?

File: src/components/header/header-modal/header-modal.tsx [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+  const styles = useHeaderModalStyles();
+  const dispatch = useDispatch();
+
+  const selectedAccount = useSelectedAccountSelector();
+  const visibleAccounts = useVisibleAccountsListSelector();
+
+  return (
+    <View style={styles.accountContainer}>
+      <CurrentAccountDropdown
+        isModal
```
Comment (unixvb) [inline src/components/header/header-modal/header-modal.tsx:20]: please, set `boolean` values `explicitly`

File: src/components/header/header-close-options-button/header-close-options-button.tsx [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+export const HeaderCloseOptionsButton: FC = () => {
+  const styles = useHeaderCloseOptionsButtonStyles();
+  const { navigate } = useNavigation();
+
+  return (
+    <View style={styles.accountActionsContainer}>
+      <TouchableIcon
+        size={formatSize(16)}
+        name={IconNameEnum.MoreHorizontal}
+        disabled
```
Comment (unixvb) [inline src/components/header/header-close-options-button/header-close-options-button.tsx:20]: please, set `boolean` values `explicitly`

File: src/components/dropdown/dropdown.tsx [hunk 13]
```diff
@@ -69,6 +69,7 @@ export const Dropdown = <T extends unknown>({
         style={styles.valueContainer}
         onPress={dropdownBottomSheetController.open}
         onLongPress={onLongPress}
+        disabled={disabled}
```
Comment (unixvb) [inline src/components/dropdown/dropdown.tsx:72]: props order

File: src/components/account-dropdown/current-account-dropdown.tsx [hunk 14]
```diff
@@ -62,7 +67,7 @@ export const CurrentAccountDropdown: FC<DropdownValueProps<WalletAccountInterfac
       value={value}
       list={list}
       equalityFn={accountEqualityFn}
-      renderValue={renderAccountValue}
+      renderValue={isModal === true ? renderAccountValueWithModal : renderAccountValue}
```
Comment (unixvb) [inline src/components/account-dropdown/current-account-dropdown.tsx:70]: please, do not use conditional rendering just create 2 different components

File: src/components/account-dropdown/modal-account-dropdown.tsx [hunk 15]
```diff
@@ -0,0 +1,26 @@
+import React, { FC } from 'react';
+
+import { WalletAccountInterface } from '../../interfaces/wallet-account.interface';
+import { DropdownValueComponent, DropdownValueProps } from '../dropdown/dropdown';
+import { IconNameEnum } from '../icon/icon-name.enum';
+import { AccountDropdownBase } from './account-dropdown-base';
+import { AccountDropdownItem } from './account-dropdown-item/account-dropdown-item';
+
+const renderAccountValueWithModal: DropdownValueComponent<WalletAccountInterface> = ({ value }) => (
+  <AccountDropdownItem isModal account={value} showFullData={false} actionIconName={IconNameEnum.TriangleDown} />
```
Comment (unixvb) [inline src/components/account-dropdown/modal-account-dropdown.tsx:10]: split into 2 components

**Review by unixvb — APPROVED** [review]

**Review by LendiHop — APPROVED** [review]

**Pull request merged**